lib/reply.js

```diff
@@ -249,8 +249,12 @@ function sendStream (payload, res, reply) {
   // appropriately, e.g. a 404 for a missing file. So we cannot use
   // writeHead, and we need to resort to setHeader, which will trigger
   // a writeHead when there is data to send.
-  for (var key in reply._headers) {
-    res.setHeader(key, reply._headers[key])
+  if (!res.headersSent) {
+    for (var key in reply._headers) {
+      res.setHeader(key, reply._headers[key])
+    }
+  } else {
+    res.log.warn('response will send, but you shouldn\'t use res.writeHead in stream mode')
```

Comment:
I would rephrase this as:`res.log.warn('Headers already sent, the following headers will not be sent', reply._headers)`

Reply:
I don't think so.   
Because when `response.headersSent === true`, it denotes the header has been send. Either `reply._headers` or `response.getHeader` can't get the value send by `reply.res.writeHead`

Reply:
So this case could happen **only** if you are using `writeHead`?

Reply:
Yes :)